vclib-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidc/helpers/ResponseModeParameters.kt

```diff
@@ -0,0 +1,33 @@
+package at.asitplus.wallet.lib.oidc.helpers
+
+import at.asitplus.wallet.lib.oidc.OpenIdConstants
+import kotlinx.serialization.Serializable
+
+@Serializable
+sealed interface ResponseModeParameters {
+    val responseMode: OpenIdConstants.ResponseMode
+
+    @Serializable
+    class DirectPost(val responseUrl: String) : ResponseModeParameters {
+        override val responseMode: OpenIdConstants.ResponseMode
+            get() = OpenIdConstants.ResponseMode.DIRECT_POST
```

Comment:
Is this really needed, since we "know" the response mode from name of this class. Otherwise, couldn't we put a member into the base class with the value of `parameters.params.responseMode`?

Reply:
> Is this really needed, since we "know" the response mode from name of this class
I'm not sure when something is "needed", there may as well be other solutions

> Otherwise, couldn't we put a member into the base class with the value of parameters.params.responseMode?
Yes, but this would add an additional backing field that's not really required as, as you pointed out above, we "know" the response mode from name of this class.